hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaSoftwareModuleManagement.java

```diff
@@ -202,19 +205,23 @@ public Optional<SoftwareModule> get(final long id) {
     @Override
     public Optional<SoftwareModule> getByNameAndVersionAndType(final String name, final String version,
             final long typeId) {
+        assertSoftwareModuleTypeExists(typeId);
 
-        throwExceptionIfSoftwareModuleTypeDoesNotExist(typeId);
-
-        return softwareModuleRepository.findOneByNameAndVersionAndTypeId(name, version, typeId);
-    }
-
-    private boolean isUnassigned(final Long moduleId) {
-        return distributionSetRepository.countByModulesId(moduleId) <= 0;
+        // TODO AC : this method is used to validate input. Is it fine to restrict the
+        // access?
```

Comment:
Ok - access check will stay here. This could have problems with Vaadin UI when access control is enabled.
However, probably Vaadin UI will be removed in few months (non from community supports its migration to new Vaadin versions), probably no problem. If, Vaadin UI stay for more time we could add correspondent permissions elevation for UI if needed